hbase-balancer/src/main/java/org/apache/hadoop/hbase/master/balancer/StochasticLoadBalancer.java

```diff
@@ -166,37 +178,37 @@ public StochasticLoadBalancer(MetricsStochasticBalancer metricsStochasticBalance
     super(metricsStochasticBalancer);
   }
 
-  private static CostFunction createCostFunction(Class<? extends CostFunction> clazz,
-    Configuration conf) {
-    try {
-      Constructor<? extends CostFunction> ctor = clazz.getDeclaredConstructor(Configuration.class);
-      return ReflectionUtils.instantiate(clazz.getName(), ctor, conf);
-    } catch (NoSuchMethodException e) {
-      // will try construct with no parameter
-    }
-    return ReflectionUtils.newInstance(clazz);
-  }
-
-  private void loadCustomCostFunctions(Configuration conf) {
-    String[] functionsNames = conf.getStrings(COST_FUNCTIONS_COST_FUNCTIONS_KEY);
+  private <T> List<? extends T> loadFunctions(Configuration conf, String configName) {
```

Comment:
Why change to T instead of CostFunction here?

Reply:
I reused this function to load the ClusterStatusBalanceChecker classes, which I designed as a chain of checkers to confirm if the balancer needs coarse costs calculations. Though currently I only add a new OverallSkewChecker, I think there are more possibilities, e.g. OverallHighLoadChecker, which can choose those high load tables and balance them first using the coarse cost functions. Thanks.